test/catalog/services/test_udf_cost_catalog_service.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest import TestCase
+
+from mock import patch
+from sqlalchemy.orm.exc import NoResultFound
+
+from eva.catalog.services.udf_cost_catalog_service import UdfCostCatalogService
+
+UDF_TYPE = "classification"
+UDF_NAME = "name"
+UDF_COST = 1
+FRAME_COUNT = 1
+RESOLUTION = 1
+
+
+class UdfCostCatalogServiceTest(TestCase):
```

Comment:
Also, add an e2e test case, run a query with UDF, and then retrieve the cost from the catalog table.

Reply:
I would prefer to do this once the cost actually gets used in the optimizer, for now it seems to be a large code delta in the testing, that doesn't impact coverage as much